automation/capitanodoc/capitanodoc.ts

```diff
@@ -27,224 +28,106 @@ import { MarkdownFileParser } from './utils';
-				'build/commands/fleet/rename.js',
-				'build/commands/fleet/restart.js',
-				'build/commands/fleet/rm.js',
-				'build/commands/fleet/track-latest.js',
-			],
-		},
-		{
-			title: 'Fleets',
-			files: ['build/commands/fleets.js'],
-		},
-		{
-			title: 'Help and Version',
-			files: ['help', 'build/commands/version.js'],
-		},
-		{
-			title: 'Local',
-			files: [
-				'build/commands/local/configure.js',
-				'build/commands/local/flash.js',
-			],
-		},
-		{
-			title: 'Logs',
-			files: ['build/commands/logs.js'],
-		},
-		{
-			title: 'Network',
-			files: [
-				'build/commands/scan.js',
-				'build/commands/ssh.js',
-				'build/commands/tunnel.js',
-			],
-		},
-		{
-			title: 'Notes',
-			files: ['build/commands/note.js'],
-		},
-		{
-			title: 'Organizations',
-			files: ['build/commands/orgs.js'],
-		},
-		{
-			title: 'OS',
-			files: [
-				'build/commands/os/build-config.js',
-				'build/commands/os/configure.js',
-				'build/commands/os/download.js',
-				'build/commands/os/initialize.js',
-				'build/commands/os/versions.js',
-			],
-		},
-		{
-			title: 'Preload',
-			files: ['build/commands/preload.js'],
-		},
-		{
-			title: 'Push',
-			files: ['build/commands/push.js'],
-		},
-		{
-			title: 'Platform',
-			files: ['build/commands/join.js', 'build/commands/leave.js'],
-		},
-		{
-			title: 'Release',
-			files: [
-				'build/commands/release/finalize.js',
-				'build/commands/release/index.js',
-				'build/commands/release/invalidate.js',
-				'build/commands/release/validate.js',
-			],
-		},
-		{
-			title: 'Releases',
-			files: ['build/commands/releases.js'],
-		},
-		{
-			title: 'Settings',
-			files: ['build/commands/settings.js'],
-		},
-		{
-			title: 'Support',
-			files: ['build/commands/support.js'],
-		},
-
-		{
-			title: 'SSH Key',
-			files: [
-				'build/commands/key/add.js',
-				'build/commands/key/index.js',
-				'build/commands/key/rm.js',
-			],
-		},
-		{
-			title: 'SSH Keys',
-			files: ['build/commands/keys.js'],
-		},
-		{
-			title: 'Tags',
-			files: ['build/commands/tag/rm.js', 'build/commands/tag/set.js'],
-		},
-		{
-			title: 'Tags',
-			files: ['build/commands/tags.js'],
-		},
-		{
-			title: 'Utilities',
-			files: ['build/commands/util/available-drives.js'],
-		},
-	],
+	categories: [],
 };
 
+// Helper function to capitalize each word of directory name
+function formatTitle(dir: string): string {
+	return dir.replace(/(^\w|\s\w)/g, (word) => word.toUpperCase());
+}
+
+// Create a map to track the categories for faster lookup
+const categoriesMap: { [key: string]: Category } = {};
+
+for (const commandPath of allCommandsPaths) {
+	const commandDir = path.basename(path.dirname(commandPath));
+	const heading = commandHeadings[commandDir] || formatTitle(commandDir);
+
+	if (!categoriesMap[heading]) {
+		categoriesMap[heading] = { title: heading, files: [] };
+		capitanoDoc.categories.push(categoriesMap[heading]);
+	}
+
+	categoriesMap[heading].files.push(commandPath);
```

Comment:
Shouldn't we pushed to `categoriesMap[heading].files` first and *then* pushed to `capitanoDoc.categories`?

Reply:
You need to push an object first to `capitanoDoc.categories` having the `{title: "something", files: [ ] }`
I also was confused how exactly the `files` keeps updated on every loop instance but it seems that Node on Line 110 is really pushing a variable to `capitanoDoc.categories` than the actual value. I tried this here. 

```
➜  ~ node
> b = {a: [1]}
{ a: [ 1 ] }
> mega = {mega: b}
{ mega: { a: [ 1 ] } }
> b['a'].push(123)
2
> console.log(mega)
{ mega: { a: [ 1, 123 ] } }
```

Let me know if I have clarified incorrectly. This was a TIL for me too why this works but it does.